Can't compile Godot with minimal build

### Godot version

4.0 42ebc4ef65093ac7bb4e1785054e4632d3ddc9bc

### System information

Ubuntu 22.04 - Nvidia GTX 970, Gnome shell 42 X11

### Issue description

This is max 1 day old regression

```
scons -j2 dev_build=yes target=editor linker=gold module_basis_universal_enabled=no module_bmp_enabled=no 
module_camera_enabled=no module_csg_enabled=no module_cvtt_enabled=no module_dds_enabled=no 
module_denoise_enabled=no module_enet_enabled=no module_etcpak_enabled=no brotli=no module_gltf_enabled=no 
module_gridmap_enabled=no module_hdr_enabled=no module_jpg_enabled=no module_jsonrpc_enabled=no 
module_lightmapper_rd_enabled=no module_mbedtls_enabled=no module_meshoptimizer_enabled=no 
module_minimp3_enabled=no module_mobile_vr_enabled=no module_msdfgen_enabled=no module_multiplayer_enabled=no 
module_noise_enabled=no module_ogg_enabled=no module_openxr_enabled=no module_raycast_enabled=no 
module_regex_enabled=no module_squish_enabled=no graphite=no module_text_server_adv_enabled=no 
module_tga_enabled=no module_theora_enabled=no module_tinyexr_enabled=no module_upnp_enabled=no 
module_vhacd_enabled=no module_vorbis_enabled=no module_webp_enabled=no module_webrtc_enabled=no 
module_websocket_enabled=no module_webxr_enabled=no module_xatlas_unwrap_enabled=no
```
fails to compile with this error
```
 Compiling platform/linuxbsd/export/export_plugin.cpp ...
Compiling platform/macos/export/codesign.cpp ...
Compiling platform/macos/export/export.cpp ...
Compiling platform/macos/export/export_plugin.cpp ...
Compiling platform/macos/export/lipo.cpp ...
Compiling platform/macos/export/macho.cpp ...
platform/macos/export/export_plugin.cpp: In member function 'virtual bool EditorExportPlatformMacOS::is_executable(const String&) const':
platform/macos/export/export_plugin.cpp:909:9: error: 'MachO' has not been declared
  909 |  return MachO::is_macho(p_path) || LipO::is_lipo(p_path) || is_shbang(p_path);
      |         ^~~~~
platform/macos/export/export_plugin.cpp:909:36: error: 'LipO' has not been declared
  909 |  return MachO::is_macho(p_path) || LipO::is_lipo(p_path) || is_shbang(p_path);
      |                                    ^~~~
scons: *** [platform/macos/export/export_plugin.linuxbsd.editor.dev.x86_64.san.o] Error 1
scons: building terminated because of errors.
[Time elapsed: 00:04:38.197]
```

Looks like regression from https://github.com/godotengine/godot/pull/66952
CC @bruvzg 

### Steps to reproduce

Above

### Minimal reproduction project

_No response_

--- a/platform/macos/export/lipo.cpp
+++ b/platform/macos/export/lipo.cpp
@@ -28,11 +28,7 @@
 /* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
 /*************************************************************************/
 
-#include "modules/modules_enabled.gen.h" // For regex.
-
 #include "lipo.h"
-
-#ifdef MODULE_REGEX_ENABLED
 
 bool LipO::is_lipo(const String &p_path) {
 	Ref<FileAccess> fb = FileAccess::open(p_path, FileAccess::READ);
@@ -232,5 +228,3 @@ void LipO::close() {
 LipO::~LipO() {
 	close();
 }
-
-#endif // MODULE_REGEX_ENABLED

--- a/platform/macos/export/lipo.h
+++ b/platform/macos/export/lipo.h
@@ -35,11 +35,8 @@
 
 #include "core/io/file_access.h"
 #include "core/object/ref_counted.h"
-#include "modules/modules_enabled.gen.h" // For regex.
 
 #include "macho.h"
-
-#ifdef MODULE_REGEX_ENABLED
 
 class LipO : public RefCounted {
 	struct FatArch {
@@ -70,7 +67,5 @@ public:
 
 	~LipO();
 };
-
-#endif // MODULE_REGEX_ENABLED
 
 #endif // MACOS_LIPO_H

--- a/platform/macos/export/macho.cpp
+++ b/platform/macos/export/macho.cpp
@@ -28,11 +28,7 @@
 /* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
 /*************************************************************************/
 
-#include "modules/modules_enabled.gen.h" // For regex.
-
 #include "macho.h"
-
-#ifdef MODULE_REGEX_ENABLED
 
 uint32_t MachO::seg_align(uint64_t p_vmaddr, uint32_t p_min, uint32_t p_max) {
 	uint32_t salign = p_max;
@@ -544,5 +540,3 @@ bool MachO::set_signature_size(uint64_t p_size) {
 	}
 	return true;
 }
-
-#endif // MODULE_REGEX_ENABLED

--- a/platform/macos/export/macho.h
+++ b/platform/macos/export/macho.h
@@ -37,9 +37,6 @@
 #include "core/crypto/crypto_core.h"
 #include "core/io/file_access.h"
 #include "core/object/ref_counted.h"
-#include "modules/modules_enabled.gen.h" // For regex.
-
-#ifdef MODULE_REGEX_ENABLED
 
 class MachO : public RefCounted {
 	struct MachHeader {
@@ -209,7 +206,5 @@ public:
 	uint64_t get_signature_size();
 	bool set_signature_size(uint64_t p_size);
 };
-
-#endif // MODULE_REGEX_ENABLED
 
 #endif // MACOS_MACHO_H

--- a/platform/macos/export/plist.cpp
+++ b/platform/macos/export/plist.cpp
@@ -28,11 +28,7 @@
 /* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
 /*************************************************************************/
 
-#include "modules/modules_enabled.gen.h" // For regex.
-
 #include "plist.h"
-
-#ifdef MODULE_REGEX_ENABLED
 
 Ref<PListNode> PListNode::new_array() {
 	Ref<PListNode> node = memnew(PListNode());
@@ -566,5 +562,3 @@ String PList::save_text() const {
 Ref<PListNode> PList::get_root() {
 	return root;
 }
-
-#endif // MODULE_REGEX_ENABLED

--- a/platform/macos/export/plist.h
+++ b/platform/macos/export/plist.h
@@ -35,9 +35,6 @@
 
 #include "core/crypto/crypto_core.h"
 #include "core/io/file_access.h"
-#include "modules/modules_enabled.gen.h" // For regex.
-
-#ifdef MODULE_REGEX_ENABLED
 
 class PListNode;
 
@@ -110,7 +107,5 @@ public:
 	PListNode() {}
 	~PListNode() {}
 };
-
-#endif // MODULE_REGEX_ENABLED
 
 #endif // MACOS_PLIST_H

